server/elastic-server/src/main/java/com/rbmhtechnology/vind/elasticsearch/backend/ElasticsearchServerProvider.java

```diff
@@ -26,12 +30,28 @@ public ElasticVindClient getInstance() {
         log.info("Instantiating Elasticsearch client: {}", host);
 
         if(collection != null) {
-            ElasticVindClient client =
-                    new ElasticVindClient.Builder(host)
+            ElasticVindClient client;
+            final AuthTypes authType = valueOf(SearchConfiguration.get(SearchConfiguration.SEARCH_AUTHENTICATION_METHOD, NONE.name()));
+            switch (authType) {
+                case APIKEY:
+                    client = new ElasticVindClient.Builder(host)
+                            .setDefaultIndex(collection)
+                            .build(SearchConfiguration.get(SearchConfiguration.SEARCH_API_KEY_AUTH));
+                    break;
+                case BASIC:
+                    client = new ElasticVindClient.Builder(host)
+                                .setDefaultIndex(collection)
+                                .build(
+                                        SearchConfiguration.get(SearchConfiguration.SEARCH_AUTHENTICATION_USER),
```

Comment:
do a pre-condition check here as it is also not checked in the builder / constructor as far as I can see. so possible NPE